crates/protocol/src/utils.rs

```diff
@@ -224,9 +231,85 @@ fn u24(input: &[u8], idx: u32) -> u32 {
         + (u32::from(input[(idx + 2) as usize]) << 16)
 }
 
+/// Reads transaction data from a reader.
+#[allow(unused)]
+pub fn read_tx_data(r: &mut &[u8]) -> Result<(Vec<u8>, TxType), SpanBatchError> {
+    let mut tx_data = Vec::new();
+    let first_byte =
+        *r.first().ok_or(SpanBatchError::Decoding(SpanDecodingError::InvalidTransactionData))?;
```

Comment:
we can just use `Buf::get_u8()` and similar here instead of manually doing `.first()` and `advance`.

Reply:
I don't think so since we only advance if the tx type is legacy, otherwise we need to leave the first byte to decode the header. Maybe I can put up a test to demonstrate this